File: lib/features/support/domain/support_ticket.dart
/// Обращение в поддержку: тема, статус и ответ, когда он появится.
class SupportTicket {
  const SupportTicket({
    required this.id,
    required this.subject,
    required this.status,
    this.createdAt,
    this.answer,
    this.answeredAt,
  });

  final String id;
  final String subject;

  /// `open` — ждёт разбора, `answered` — есть ответ.
  final String status;
  final DateTime? createdAt;
  final String? answer;
  final DateTime? answeredAt;

  bool get isAnswered => status == 'answered';

  static SupportTicket? fromMap(Map<String, dynamic> map) {
    final id = map['id']?.toString();
    final subject = map['subject']?.toString();
    if (id == null || subject == null) return null;
    return SupportTicket(
      id: id,
      subject: subject,
      status: map['status']?.toString() ?? 'open',
      createdAt: DateTime.tryParse(map['createdAt']?.toString() ?? ''),
      answer: map['answer']?.toString(),
      answeredAt: DateTime.tryParse(map['answeredAt']?.toString() ?? ''),
    );
  }
}

File: aloria-api/src/Aloria.Api/Domain/SupportTicket.cs
namespace Aloria.Api.Domain;

/// <summary>
/// Обращение в поддержку. Создаётся из приложения, когда пользователь
/// столкнулся с системным сбоем («проблема в мире Алории») или хочет
/// сообщить о проблеме. Вместе с обращением сохраняется подробный контекст
/// (параметры заявки, портфель, позиции) для разбора без переписки.
/// </summary>
public class SupportTicket
{
    public Guid Id { get; set; }

    public Guid UserId { get; set; }
    public User? User { get; set; }

    /// Статус: open — ждёт разбора, answered — есть ответ.
    public string Status { get; set; } = "open";

    /// Короткая тема (например, «Не отправилась заявка по SBER»).
    public string Subject { get; set; } = string.Empty;

    /// Код ошибки торговой системы, если был (например, CommandResponseTimeout).
    public string? ErrorCode { get; set; }

    /// Техническое сообщение ошибки.
    public string? ErrorMessage { get; set; }

    /// JSON-снимок контекста: параметры заявки, покупательная способность,
    /// позиции, версия приложения и т.п.
    public string? ContextJson { get; set; }

    /// Комментарий пользователя своими словами (необязательный).
    public string? UserComment { get; set; }

    /// Ответ поддержки (заполняется через админку).
    public string? Answer { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? AnsweredAt { get; set; }
}
